packages/components/src/tabs/styles.ts

```diff
@@ -38,20 +38,41 @@ export const TabListWrapper = styled.div`
 		outline: 2px solid transparent;
 		outline-offset: -1px;
 	}
-	&:not( [aria-orientation='vertical'] )::after {
-		bottom: 0;
-		left: var( --indicator-left );
-		width: var( --indicator-width );
-		height: 0;
-		border-bottom: var( --wp-admin-border-width-focus ) solid
-			${ COLORS.theme.accent };
+	&:not( [aria-orientation='vertical'] ) {
+		--direction-factor: 1;
+		--indicator-start: var( --indicator-left );
+		&:dir( rtl ) {
+			--direction-factor: -1;
+			--indicator-start: var( --indicator-right );
+		}
+		&::after {
+			bottom: 0;
+			height: 0;
+			/* Using a large value to avoid antialiasing rounding issues
+			when scaling in the transform, see: https://stackoverflow.com/a/52159123 */
+			width: 100px;
+			transform-origin: calc( ( 1 - var( --direction-factor ) ) * 50% ) 0%;
+			transform: translateX(
+					calc(
+						var( --indicator-start ) * var( --direction-factor ) *
+							1px
+					)
+				)
+				scaleX( calc( var( --indicator-width ) / 100 ) );
+			border-bottom: var( --wp-admin-border-width-focus ) solid
+				${ COLORS.theme.accent };
+		}
 	}
 	&[aria-orientation='vertical']::after {
 		z-index: -1;
+		top: 0;
 		left: 0;
 		width: 100%;
-		top: var( --indicator-top );
-		height: var( --indicator-height );
+		/* Using a large value to avoid antialiasing rounding issues
+			when scaling in the transform, see: https://stackoverflow.com/a/52159123 */
+		height: 100px;
+		transform: translateY( calc( var( --indicator-top ) * 1px ) )
+			scaleY( calc( var( --indicator-height ) / 100 ) );
```

Comment:
```suggestion
		transform: translateY( calc( var( --indicator-top ) * 1px ) )
			scaleY( calc( var( --indicator-height ) / 100 ) );
		transform-origin: top;
```

In a vertical layout, I think we need to start from the top. Otherwise, the background color will be misaligned:

![image](https://github.com/user-attachments/assets/3e68058b-72b8-4403-b5ae-f8e7431b0564)

```

Reply:
We start from the top already. Can't repro this.

Reply:
Strangely enough, I can reproduce this in my Storybook. The initial value of `transform-origin` is `50% 50% 0`, and `transform-origin` does not seem to be applied 🤔

![tabs](https://github.com/user-attachments/assets/199ede36-b02a-411e-a1da-fc4c37f74cda)

![image](https://github.com/user-attachments/assets/943662eb-fee9-4795-84cb-6a4fdab330b9)

I can also see the problem in the Preferences panel of the block editor:

Reply:
Are you sure you're in the latest version of the branch? Can you check that you have the `transform-origin: calc( ( 1 - var( --direction-factor ) ) * 50% ) 0%;` like in the styles source? Can you investigate whether it's applied correctly? Depending on the `--direction-factor` variable, the result for the first parameter (X axis) should be 0% (LTR) or 100% (RTL), and the second (Y axis) should always be 0% since it's a static value.

Reply:
Wait, I managed to repro. Weird, let me take a look.

Reply:
Alright, it was a dumb mistake I made haha. I placed the origin only in the horizontal tabs styles instead of in the more global scope. Fixed now!